lib/ExpensiMark.ts

```diff
@@ -1079,46 +1131,40 @@ export default class ExpensiMark {
     }
 
     /**
-     * @param {String} comment
-     * @returns {Array} or undefined if exception occurs when executing regex matching
+     * @returns or undefined if exception occurs when executing regex matching
      */
-    extractLinksInMarkdownComment(comment) {
+    extractLinksInMarkdownComment(comment: string): string[] | undefined {
         try {
             const htmlString = this.replace(comment, {filterRules: ['link']});
             // We use same anchor tag template as link and autolink rules to extract link
             const regex = new RegExp(`<a href="${UrlPatterns.MARKDOWN_URL_REGEX}" target="_blank" rel="noreferrer noopener">`, 'gi');
             const matches = [...htmlString.matchAll(regex)];
 
             // Element 1 from match is the regex group if it exists which contains the link URLs
-            const sanitizeMatch = (match) => Str.sanitizeURL(match[1]);
+            const sanitizeMatch = (match: RegExpExecArray) => Str.sanitizeURL(match[1]);
             const links = matches.map(sanitizeMatch);
             return links;
         } catch (e) {
-            // eslint-disable-next-line no-console
-            console.warn('Error parsing url in ExpensiMark.extractLinksInMarkdownComment', {error: e});
+            ExpensiMark.Log.alert('Error parsing url in ExpensiMark.extractLinksInMarkdownComment', {error: e});
             return undefined;
         }
     }
 
     /**
      * Compares two markdown comments and returns a list of the links removed in a new comment.
-     *
-     * @param {String} oldComment
-     * @param {String} newComment
-     * @returns {Array}
      */
-    getRemovedMarkdownLinks(oldComment, newComment) {
+    getRemovedMarkdownLinks(oldComment: string, newComment: string): string[] {
         const linksInOld = this.extractLinksInMarkdownComment(oldComment);
         const linksInNew = this.extractLinksInMarkdownComment(newComment);
         return linksInOld === undefined || linksInNew === undefined ? [] : linksInOld.filter((link) => !linksInNew.includes(link));
     }
 
     /**
      * Escapes the content of an HTML attribute value
-     * @param {String} content - string content that possible contains HTML
-     * @returns {String} - original MD content escaped for use in HTML attribute value
+     * @param content - string content that possible contains HTML
+     * @returns - original MD content escaped for use in HTML attribute value
```

Comment:
```suggestion
     * @returns original MD content escaped for use in HTML attribute value
```

Also what is MD? Markdown?